inbm/dispatcher-agent/dispatcher/sota/oras_util.py

```diff
@@ -0,0 +1,150 @@
+    """Downloads files and places capsule file in path mentioned by manifest file.
+
+    @param dispatcher_broker: DispatcherBroker object used to communicate with other INBM services
+    @param uri: URI of the source location
+    @param repo: repository for holding the download
+    @param username: username to use for download
+    @param password: password to use for download
+    @param umask: file permission mask
+    @raises DispatcherException: any exception
+    """
+    dispatcher_broker.telemetry(f'Package to be fetched from {uri.value}')
+    dispatcher_broker.telemetry(
+        'Checking authenticity of package by checking signature and source')
+
+    if not isinstance(uri, CanonicalUri):
+        raise DispatcherException("Internal error: uri improperly passed to download function")
+
+    """
+    In case of uri.value = https://registry-rs.internal.ledgepark.intel.com/one-intel-edge/tiberos:latest
+    source = https://registry-rs.internal.ledgepark.intel.com/one-intel-edge
+    registry_server = registry-rs.internal.ledgepark.intel.com
+    image = tiberos
+    image_tag = latest
+    image_full_path = registry-rs.internal.ledgepark.intel.com/one-intel-edge/tiberos:latest
+    repository_name = one-intel-edge
+    registry_manifest = https://registry-rs.internal.ledgepark.intel.com/v2/one-intel-edge/tiberos/manifest/latest
+    """
+    try:
+        source = uri.value[:-(len(uri.value.split('/')[-1]) + 1)]
+        registry_server = uri.value.split("/")[2]
+        image = os.path.basename(urlsplit(uri.value).path).split(":")[0]
+        image_tag = uri.value.split(":")[-1]
+        repository_name = "/".join(uri.value.split("/")[3:-1])
+        image_full_path = f"{registry_server}/{repository_name}/{image}:{image_tag}"
+        registry_manifest = f"https://{registry_server}/v2/{repository_name}/{image}/manifests/{image_tag}"
+    except IndexError as err:
+        logger.error(f"IndexError occurs with uri {uri.value}: {err}")
+        raise DispatcherException(err)
+
+    logger.debug(f"source: {source}, "
+                 f"registry_server: {registry_server}, "
+                 f"image: {image}, "
+                 f"image_tag: {image_tag}, "
+                 f"repository_name: {repository_name}, "
+                 f"registry_manifest: {registry_manifest}")
+
+    verify_source(source=source, dispatcher_broker=dispatcher_broker)
+    dispatcher_broker.telemetry('Source Verification check passed')
+
+    try:
+        enough_space = is_enough_space_to_download(
+            registry_manifest, repo, password)
+    except DispatcherException as e:
+        raise DispatcherException(e)
+
+    if not enough_space:
+        err_msg = " Insufficient free space available on " + repo.get_repo_path() + \
+                  " for " + str(uri.value)
+        raise DispatcherException(err_msg)
+
+    if password:
+        logger.debug("JWT Token provided.")
```

Comment:
confusing--argument is `password`, log message refers to `JWT Token`

Reply:
The oras uses password as the argument to pass the JWT token. When INBM run the ORAS command, it is `oras pull <url>  --password <JWT token>`

Reply:
> The oras uses password as the argument to pass the JWT token. When INBM run the ORAS command, it is `oras pull <url> --password <JWT token>`

IMO the debug statement should match the name of the argument, so either the debug statement should change or the argument should.

Reply:
Will change the debug statement.